web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -0,0 +1,421 @@
+    if (fetchClustersAttempt.status === 'failed') {
+      setFetchClustersAttempt({ status: '' });
+    }
+    if (tableData.items.length > 0) {
+      setTableData(emptyTableData);
+    }
+    if (selectedCluster) {
+      setSelectedCluster(null);
+    }
+    setEnrollmentState({ status: 'notStarted' });
+  }
+
+  function fetchKubeServers(query: string, limit: number) {
+    return ctx.kubeService.fetchKubernetes(ctx.storeUser.getClusterId(), {
+      query,
+      limit,
+    });
+  }
+
+  async function handleOnEnroll() {
+    const integrationName = (agentMeta as EksMeta).awsIntegration.name;
+    setEnrollmentState({ status: 'enrolling' });
+
+    try {
+      const response = await integrationService.enrollEksClusters(
+        integrationName,
+        {
+          region: selectedRegion,
+          enableAppDiscovery: isAppDiscoveryEnabled,
+          joinToken: joinToken.id,
+          resourceId: joinToken.internalResourceId,
+          clusterNames: [selectedCluster.name],
+        }
+      );
+
+      const result = response.results?.find(
+        c => c.clusterName === selectedCluster.name
+      );
+      if (!result) {
+        setEnrollmentState({
+          status: 'error',
+          error: `Cluster "${selectedCluster.name}" enrollment result is unknown.`,
+        });
+      } else if (result.error) {
+        setEnrollmentState({
+          status: 'error',
+          error: `Cluster enrollment error: ${result.error}`,
+        });
+      } else {
+        setEnrollmentState({ status: 'awaitingAgent' });
+        setIsAgentWaitingDialogShown(true);
+      }
+    } catch (err) {
+      setEnrollmentState({
+        status: 'error',
+        error: `Cluster enrollment error: ${getErrMessage(err)}.`,
+      });
+    }
+  }
+
+  async function handleOnProceed() {
+    props.updateAgentMeta({
+      kube: confirmedCluster,
+      resourceName: confirmedCluster.name,
+    } as EksMeta);
+
+    props.nextStep();
+  }
+
+  const hasIamPermError = isIamPermError(fetchClustersAttempt);
+
+  const closeEnrollmentDialog = () => {
+    setEnrollmentState({ status: 'notStarted' });
+  };
+
+  const enrollmentNotAllowed =
+    fetchClustersAttempt.status === 'processing' ||
+    !selectedCluster ||
+    enrollmentState.status !== 'notStarted';
+
+  let command = '';
+  if (selectedCluster) {
+    command = generateCmd({
+      namespace: 'teleport-agent',
+      clusterName: selectedCluster.name,
+      proxyAddr: ctx.storeUser.state.cluster.publicURL,
+      tokenId: joinToken.id,
+      clusterVersion: ctx.storeUser.state.cluster.authVersion,
+      resourceId: joinToken.internalResourceId,
+      isEnterprise: ctx.isEnterprise,
+      isCloud: ctx.isCloud,
+      automaticUpgradesEnabled: ctx.automaticUpgradesEnabled,
+      automaticUpgradesTargetVersion: ctx.automaticUpgradesTargetVersion,
+      joinLabels: [...selectedCluster.labels, ...selectedCluster.joinLabels],
+      disableAppDiscovery: !isAppDiscoveryEnabled,
+    });
+  }
```

Comment:
this looks like it belongs in the `ManualHelmDialog.tsx`

Reply:
I actually like it better if `ManualHelmDialog` is just happily showing any text we give it. To me it seems like a more logical place and we don't need to pass a lot of parameters into the dialog.